wrangler-service/src/main/java/io/cdap/wrangler/service/adls/ADLSHandler.java

```diff
@@ -0,0 +1,382 @@
+import io.cdap.wrangler.dataset.workspace.DataType;
+import io.cdap.wrangler.dataset.workspace.WorkspaceDataset;
+import io.cdap.wrangler.dataset.workspace.WorkspaceMeta;
+import io.cdap.wrangler.proto.BadRequestException;
+import io.cdap.wrangler.proto.*;
+import io.cdap.wrangler.proto.adls.ADLSConnectionSample;
+import io.cdap.wrangler.proto.adls.ADLSDirectoryEntryInfo;
+import io.cdap.wrangler.proto.adls.FileQueryDetails;
+import io.cdap.wrangler.proto.connection.Connection;
+import io.cdap.wrangler.proto.connection.ConnectionMeta;
+import io.cdap.wrangler.proto.connection.ConnectionType;
+import io.cdap.wrangler.sampling.Bernoulli;
+import io.cdap.wrangler.sampling.Poisson;
+import io.cdap.wrangler.sampling.Reservoir;
+import io.cdap.wrangler.service.FileTypeDetector;
+import io.cdap.wrangler.service.common.AbstractWranglerHandler;
+import io.cdap.wrangler.service.common.Format;
+import io.cdap.wrangler.service.explorer.BoundedLineInputStream;
+import io.cdap.wrangler.utils.ObjectSerDe;
+
+import javax.annotation.Nullable;
+import javax.ws.rs.*;
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+
+/**
+ * Service to explore ADLS Gen1 filesystem.
+ */
+public class ADLSHandler extends AbstractWranglerHandler {
+    private static final String COLUMN_NAME = "body";
+    private static final int FILE_SIZE = 10 * 1024 * 1024;
+    private static final FileTypeDetector detector = new FileTypeDetector();
+
+    /**
+     * Create an ADLS client using connection details from the HTTP request.
+     *
+     * @param connection connection details from the HTTP request.
+     * @return ADLStoreClient
+     */
+    private ADLStoreClient initializeAndGetADLSClient(ConnectionMeta connection) {
+        ADLSConfiguration ADLSConfiguration = new ADLSConfiguration(connection.getProperties());
+        String authTokenEndpoint = ADLSConfiguration.getEndpointURL();
+        String clientId = ADLSConfiguration.getADLSClientID();
+        String clientKey = ADLSConfiguration.getClientKey();
+        String accountFQDN = ADLSConfiguration.getAccountFQDN();
+        AccessTokenProvider provider = new ClientCredsTokenProvider(authTokenEndpoint, clientId, clientKey);
+        return ADLStoreClient.createClient(accountFQDN, provider);
+    }
+
+    /**
+     * Tests ADLS Connection.
+     *
+     * @param request   HTTP Request handler.
+     * @param responder HTTP Response handler.
+     */
+    @POST
+    @Path("contexts/{context}/connections/adls/test")
+    @TransactionPolicy(value = TransactionControl.EXPLICIT)
+    public void testADLSConnection(HttpServiceRequest request, HttpServiceResponder responder,
+                                   @PathParam("context") String namespace) {
+        // Extract the body of the request and transform it to the Connection object
+        if (request == null) {
+            throw new IllegalArgumentException();
```

Comment:
Were you seeing this as null sometimes? It's not supposed to ever be null.

Reply:
My bad, I was thinking of keeping a null check on the body of the request but accidentally compared the whole request with null. I figure RequestExtractor keeps a check on all of that so I should remove this.